samtranslator/model/eventsources/push.py

```diff
@@ -324,6 +324,17 @@ def _depend_on_lambda_permissions(self, bucket, permission):
 
         depends_on = bucket.get("DependsOn", [])
 
+        # Validate that DependsOn is either a list or a string.
+        # If DependsOn is a list, also validate that all elements are also strings
+        check_invalid_type = not (isinstance(depends_on, string_types) or isinstance(depends_on, list)) or (
+            isinstance(depends_on, list) and not all(isinstance(el, str) for el in depends_on)
+        )
+        if check_invalid_type:
```

Comment:
The check below are checks that `depends_on` is a string. Should probably be consolidated.